website/src/app/kb/architecture/core-components/readme.mdx

```diff
@@ -0,0 +1,197 @@
+import SupportOptions from "@/components/SupportOptions";
+import NextStep from "@/components/NextStep";
+import Image from "next/image";
+import Link from "next/link";
+import Alert from "@/components/DocsAlert";
+
+# Architecture: Core Components
+
+Firezone consists of a few core components that work together in unison to
+deliver secure, reliable access wherever your workforce is located.
+
+These components will be referred to throughout the documentation, so it's
+important to understand what each one does and how they interact with each
+other.
+
+## Overview
+
+Components in Firezone can be divided into two categories:
+
+1. **Firezone-managed**: Components that we host and operate for you.
+1. **User-managed**: Components you host and/or operate yourself.
+
+<Link
+  target="_blank"
+  href="/images/kb/architecture/core-components/component-diagram.svg"
+>
+  <Image
+    src="/images/kb/architecture/core-components/component-diagram.svg"
+    alt="Firezone core component diagram"
+    width={1200}
+    height={1200}
+    className="p-16"
+  />
+</Link>
+
+Each component is described in more detail below.
+
+## Firezone-managed components
+
+These are components we host and operate in our cloud environment.
+
+### Admin portal
+
+`https://app.firezone.dev`
+
+The admin portal is the web-based interface where you configure and manage
+Firezone for your organization.
+
+It provides a single pane of glass for configuring authentication, Resources,
+Gateways, Policies, and more.
+
+The admin portal is delivered as a managed SaaS application that's load-balanced
+globally for high availability.
+
+### Control plane API
+
+`wss://api.firezone.dev`
+
+The control plane API is a WebSocket-based, realtime API that Relays, Gateways,
+and Clients connect to for configuration and policy updates. It's responsible
+for distributing WireGuard keys to relevant peers and serves as the signaling
+layer for Firezone's STUN/TURN implementation.
+
+All configuration updates made in the admin portal are pushed in realtime to
+connected Clients and Gateways so they always have up-to-date information about
+which Resources are available and which Gateways to use for connecting to them.
+This informs our approach to high availability in Firezone: if a Gateway goes
+offline, Clients will immediately discontinue using it for new connections.
```

Comment:
This may just be me, but as I'm reading these lines I feel like I'm not fully understanding how Firezone high availability works.  It kind of sounds like if a gateway goes offline the resources being served by the gateway are then also offline, regardless of whether or not there are multiple gateways running in that group.  Wondering if we should add in a small blurb talking about how they will get pushed to another gateway in the same group and clients will be updated in real-time to use the other gateway(s) ?

Reply:
Ah, further details are in the `Deploy gateways` docs. There were a few places where I had to ask myself "is this better explained here or in the relevant docs?"

I think my goal with this Architecture documentation is to provide a single place where someone evaluating Firezone can read through in one sitting to get a general feel for how we implemented the features we did, and not necessarily go too detailed into any aspect of the system to use to action off of, using other sections of the docs for that instead.

In this case, I assume `Deploy Gateways` would be where users are probably wondering how Firezone's high availability design affects how they should deploy gateways, and in the architecture document, it's more of a "this is what we have".

Perhaps I should move the detail from `Deploy Gateways` to here, and link to this doc from there.

Reply:
I did just that. Much better IMO

Reply:
Yeah, I agree with what you're saying, it's tough to figure out where something should be explained in detail vs just a quick overview.  I also realize that my initial comment didn't exactly reflect what I was really thinking.

What I was really trying to get across was that if felt like high availability was introduced as a concept in the paragraph, but then I felt like I was left hanging as to how high availability as a concept related to configuration updates being pushed from the portal to the gateway.  

I do like the link out to a specific high availability section with more detail.

What do you think if we added something like this:

```
All configuration updates made in the admin portal are pushed in realtime to connected Clients and Gateways so they always have up-to-date information about which Resources are available and which Gateways to use for connecting to them. This informs our approach to high availability in Firezone: websockets also allow the Portal to know if a Gateway goes offline.  Once an offline Gateway is detected Clients receive the new configuration, immediately discontinue using it for new connections, and existing connections will be re-established with another Gateway in the Site.
```

Reply:
Make it a bit more concise:

> This informs our approach to high availability in Firezone: if a Gateway goes
offline, the admin portal is immediately notified. When this happens, Clients
receive the new configuration, discontinue using the offline Gateway for new
connections, and re-establish existing connections with healthy Gateways in the
Site.

Reply:
👍 I like it